save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/ErrorBoundary.kt

```diff
@@ -0,0 +1,69 @@
+/**
+ * Support for [error boundaries](https://reactjs.org/docs/error-boundaries.html)
+ */
+
+package com.saveourtool.save.frontend.components
+
+import com.saveourtool.save.frontend.components.views.FallbackView
+import com.saveourtool.save.frontend.topBarComponent
+
+import csstype.ClassName
+import react.Component
+import react.PropsWithChildren
+import react.RStatics
+import react.ReactNode
+import react.State
+import react.create
+import react.dom.html.ReactHTML.div
+import react.react
+
+import kotlinx.js.jso
+
+/**
+ * State of error boundary component
+ */
+external interface ErrorBoundaryState : State {
+    /**
+     * True is there is an error in the wrapped component tree
+     */
+    var hasError: Boolean?
+}
+
+/**
+ * Component to act as React Error Boundary
+ */
+class ErrorBoundary : Component<PropsWithChildren, ErrorBoundaryState>() {
+    init {
+        state = jso {
+            hasError = false
+        }
+    }
+
+    override fun render(): ReactNode? = if (state.hasError == true) {
+        div.create {
+            className = ClassName("container-fluid")
+            topBarComponent()
+            FallbackView::class.react {
+                bigText = "Error"
```

Comment:
We do not pass an error number and error text here?

Reply:
In this case the error is usually some sort of internal stack trace, so I just add a generic text